### DISCUSSION TITLE
how to set a dojo pseudopotentials for espresso?

### DISCUSSION THREAD

**Discussion by yw-fang** [Quantum-Accelerators/quacc#2486] [General]
Hello, @tomdemeyere 

I am trying using some espresso recipes you implemented. Currently, the document of the setting of the pseudopotentials (pps) of Quantum Espresso (QE) is a bit confusing to me. I did set the `export QUACC_ESPRESSO_PSEUDO="/path/to/pseudopotentials/in/my/HPC" `in the .bashrc. In that directory, I have downloaded all the dojo pps I need (say Cu.upf, Ag. upf, etc). But QE calculation crashed by raising the errors
/path/to/pseudopotentials/in/my/HPC/Cu.paw.z_11.ld1.psl.v1.0.0-low.upf

The minimal code to reproduce the error:
```
atoms = bulk('Cu')
job = relax_job(atoms, relax_run=False)
```
The code seemed to want the sssp_1.3.0_pbe_efficiency potentials by default.
Can we force quacc to use the user defined pps through preset? Thank you very much!

**Comment by tomdemeyere** [comment]
Yes, you will need to specify the ASE keywords 'pseudopotentials' which is a dictionary linking each element to its pseudopotential filename.

Otherwise Quacc will use the default preset "sssp..." and look for these filenames.

It might be desirable to code a function that automatically look for correct filenames in the specified pseudo_dir, but there are multiple problems:

- Some libraries have multiple files for a given element, which one do we pick?
- Some libraries have very strange filenames, which might not be easy to recognize.

What we can do in your case is to give the list of the pseudo dojo dir to an AI and ask it to generate a .yaml preset file. @Andrew-S-Rosen would have to give his go and this could be added to main repository as a standard pseudo preset. It's even better if it has the recommended cutoffs

**Comment by yw-fang** [comment]
@tomdemeyere Thank you for your quick reply! Using the yaml file seems a good idea. We and our collaborators have used dojo pps for several years and the cutoff we used in high-throughput calculations have been very robust. I can generate the yaml file by myself to meet my own demand. 

One question about the used sssp_1.3.0_pbe_efficiency.yaml in quacc

```
pseudopotentials:
  Ac:
    cutoff_rho: 320.0
    cutoff_wfc: 40.0
    filename: Ac.us.z_11.ld1.psl.v1.0.0-high.upf
    md5: 3a78bb30464666358a3ec0edb47f6656
    pseudopotential: 100US
```

Is the line of "pseudopotential" used to mark the source of the pps? Are the recommended cutoff extracted from the upf directly?

**Reply by tomdemeyere** [reply]
Yes, because SSSP is more of a "collection" of libraries, this line indicate which library has been selected for this element. Note that you don't have to replicate all the information here, strictly speaking you could just specify the filename and the preset would work.

Can you have a look on this file on one of my branch (https://github.com/tomdemeyere/quacc/blob/pseudo_dojo/src/quacc/calculators/espresso/presets/nc_sr_0.5_pbe_standard.yaml), what do you think is this what you need?

**Reply by yw-fang** [reply]
Thank you! This is exactly what I need for the dojo pseudopotentials. It's sufficient for my needs. It works well for the case of Cu I tested. However, to be honest, the documentation will need further improvement so that beginners can know what to do easier.

**Reply by yw-fang** [reply]
@tomdemeyere I found I am using 0.4 version of dojo, and am not sure if the recommend cutoff is as the same as version 0.5 in the yaml file you generated. Did you get the cutoff from version 0.5 at https://www.pseudo-dojo.org/ (which is not available now)?

**Reply by tomdemeyere** [reply]
Thanks for the feedback about the documentation. This was indeed planned but never done.

Yes I took the cutoff from the 0.5 version and converted them in Ry (times 2). I will first open a PR with these and see what comes out of it.

**Reply by yw-fang** [reply]
Thanks, tag me whenever you need my help to enhance the document together.
